include/envoy/http/header_map.h

```diff
@@ -22,6 +22,10 @@ class LowerCaseString {
 
   const std::string& get() const { return string_; }
   bool operator==(const LowerCaseString& rhs) const { return string_ == rhs.string_; }
+  LowerCaseString& operator=(const LowerCaseString& rhs) {
```

Comment:
I don't know what the actual problem you are trying to solve is, but I think you shouldn't need any of it, so unclear on the actual problem.